test/test_memory.cpp

```diff
@@ -75,4 +75,144 @@ TEST_F(MemoryTest, LoadCache) {
       &fusion, actual_ts, {input}, {expected_output}, __LINE__, __FILE__);
 }
 
+class TMATest : public NVFuserTest {
+  void SetUp() override {
+    // requires Hopper or newer
+    if (!deviceMajorMinorCheck(9)) {
+      GTEST_SKIP() << "skipping tests on pre-Hopper GPUs";
+    }
+    NVFuserTest::SetUp();
+  }
+};
+
+TEST_F(TMATest, StoreCompleteTensor1D) {
```

Comment:
I prefer to do 5 tests. I need to carefully select the testing tensor shapes so that it doesn't take too long (note that the g2s part is not parallelized, and using a for loop to read tens of thousands of items one by one is not fast). Also, this would allow me to select one test and modify it, hack something, for more convenient development.